queue.go

```diff
@@ -260,6 +281,35 @@ func (q *DQue) Peek() (interface{}, error) {
 	return obj, nil
 }
 
+// BatchPeek returns a slice of up to 1-n objects without dequeueing them.
```

Comment:
Very minor nitpick.  "1-n objects" looked like "1 minus n objects" to me the first time I read it.  Maybe you could just say "1 to n objects".